File: src/my_class_pkg/scripts/ros_action_client.py
#!/usr/bin/env python3
import rospy
import actionlib
from my_class_pkg.msg
import MyActionAction,MyActionGoal,MyActionResult,MyActionFeedback
def feedback_cb(feedback):
rospy.loginfo('Progress:{}'.format(feedback.progress))
if __name__=='__main__':
rospy.init_node('my_client')
client=actionlib.SimpleActionClient('my_action',MyActionAction)
client.wait_for_server()
goal=MyActionGoal()
goal.object_name='world'
rospy.loginfo('Sendinggoal...')
client.send_goal(goal,feedback_cb=feedback_cb)
client.wait_for_result()
result=client.get_result()
if result.success:
rospy.loginfo('Actionsucceeded')
else:
rospy.loginfo('Actionfailed')

File: src/my_class_pkg/src/ros_action_client.cpp
#include<ros/ros.h>
#include<actionlib/client/simple_action_client.h>
#include<actionlib/client/terminal_state.h>
#include<my_class_pkg/MyActionAction.h>
int main(int argc,char**argv){
	ros::init(argc,argv,"cre_client");
	ros::NodeHandle nh;
	actionlib::SimpleActionClient<my_class_pkg::MyActionAction> client("my_action",true);
	ROS_INFO("Waitingforactionservertostart...");
	client.waitForServer();
	my_class_pkg::MyActionGoal goal;
	goal.object_name="world";
	ROS_INFO("Sendinggoal...");
	client.sendGoal(goal);
	bool finished_before_timeout=client.waitForResult(ros::Duration(30.0));
	if (finished_before_timeout){
		actionlib::SimpleClientGoalState state=client.getState();
		ROS_INFO("Actionfinished:%s",state.toString().c_str());
	}else{
		ROS_INFO("Actiondidnotfinishbeforethetimeout.");
	}
	return 0;
}
